demo/admin/src/pages/PageContentBlock.tsx

```diff
@@ -20,4 +40,73 @@ export const PageContentBlock = createBlocksBlock({
         linkList: LinkListBlock,
         fullWidthImage: FullWidthImageBlock,
     },
+    additionalFields: {
+        userGroup: {
+            defaultValue: "All",
+        },
+    },
+    AdditionalContextMenuItems: ({ block, onChange, onMenuClose }) => {
+        const [internalValue, setInternalValue] = React.useState(block.userGroup);
+        const [dialogOpen, setDialogOpen] = React.useState(false);
+        return (
+            <>
+                <MenuItem
+                    onClick={() => {
+                        setDialogOpen(true);
+                    }}
+                >
+                    <ListItemIcon>
+                        <Account />
+                    </ListItemIcon>
+                    <FormattedMessage id="cometDemo.pageContentBlock.userGroup.menuItem" defaultMessage="Visibility rules" />
+                </MenuItem>
+                <Dialog
+                    open={dialogOpen}
+                    onClose={() => {
+                        setDialogOpen(false);
+                        onMenuClose();
+                    }}
+                >
+                    <DialogTitle>
+                        <FormattedMessage id="cometDemo.pageContentBlock.userGroup.dialogTitle" defaultMessage="Visibility rules" />
+                    </DialogTitle>
+                    <DialogContent>
+                        <Select value={internalValue} onChange={(event) => setInternalValue(event.target.value)} fullWidth>
```

Comment:
why not use final-form here?

Reply:
I've actually not thought about using it, as it's only a single select. But you're right, this is better than keeping track of an internal state

Reply:
Now using Final Form: d753ae7a17106d5ddabf9eb00d7f002015139dda